module/observationSubmissions/helper.js

```diff
@@ -135,6 +135,12 @@ module.exports = class ObservationSubmissionsHelper {
                 observationSubmissionsDocument['entityTypeId'] = entityTypeDocumentsAPICall.data[0]._id;
               }
 
+              if(observationSubmissionsDocument.entityInformation.externalId){
+                let entityDetailResponse = await entityManagementService.findEntityDetails(observationSubmissionsDocument.tenantId,observationSubmissionsDocument.entityInformation.externalId)
```

Comment:
@borkarsaish65 can we use the common function like project service ?

Reply:
done